src/utilities/mdframe/handler_netcdf.cpp

```diff
@@ -0,0 +1,78 @@
+#if NETCDF_ACTIVE
+
+#include <netcdf>
+
+#include "mdframe/mdframe.hpp"
+
+namespace ngen {
+
+namespace visitors {
+
+struct mdarray_netcdf_putvar : public boost::static_visitor<void>
+{
+
+    template<typename T>
+    void operator()(const mdarray<T>& arr, netCDF::NcVar& var)
+    {
+        typename mdarray<T>::size_type rank = arr.rank();
+        std::vector<typename mdarray<T>::size_type> expanded_index(rank);
+        
+        for (auto it = arr.begin(); it != arr.end(); it++) {
+            it.mdindex(expanded_index);
+            var.putVar(expanded_index, *it);
+        }
+    }
+
+};
+
+} // namespace visitors
+
+void mdframe::to_netcdf(const std::string& path) const
+{
+    netCDF::NcFile output{path, netCDF::NcFile::replace};
+
+    std::unordered_map<std::string, netCDF::NcDim> dimmap;
+    std::unordered_map<std::string, netCDF::NcVar> varmap;
+
+    for (const auto& dim : this->m_dimensions)
+        dimmap[dim.name()] = output.addDim(dim.name(), dim.size());
+
+    for (const auto& pair : this->m_variables) {
+        netCDF::NcType* type = nullptr;
```

Comment:
What's with `type` being a pointer? All the use looks suitable for just being `netCDF::NcType`.

Its documentation looks perfectly safe to copy those around; is there some subtlety to `netCDF::NcFile::addVar` that distinguishes instances of these type objects, such that a copy of `ncInt` isn't equivalent to the original?

Reply:
(I've only ever use netCDF from Fortran code)

Reply:
This was so I could set `type` once instead of having an `if-elseif` branch that called `addVar` in each branch. `NcType` is polymorphic, and `type` here is set to point to the already initialized version of the different `NcTypes` (i.e. see `netCDF::NcInt` vs `netCDF::ncInt`).

Reply:
Ok, I see now. Assigning an arbitrary value of `NcType type` might not be valid code. Thanks.